crates/e2e/tests/e2e/protocol_fee.rs

```diff
@@ -16,413 +21,426 @@ use {
+        market_order_token.address(),
         OrderKind::Sell,
-        Some(OrderCreationAppData::Full {
-            full: json!({
-                "version": "1.1.0",
-                "metadata": {
-                    "partnerFee": {
-                        "bps":1000,
-                        "recipient": "0xb6BAd41ae76A11D10f7b0E664C5007b908bC77C9",
-                    }
-                }
-            })
-            .to_string(),
-        }),
-        100165388404261365u128.into(),
-        98879067931768848u128.into(),
+        sell_amount,
+        model::time::now_in_epoch_seconds() + 300,
     )
-    .await;
-}
+    .await
+    .unwrap()
+    .quote;
+    let partner_fee_quote_after = get_quote(
+        &services,
+        onchain.contracts().weth.address(),
+        partner_fee_order_token.address(),
+        OrderKind::Sell,
+        sell_amount,
+        model::time::now_in_epoch_seconds() + 300,
+    )
+    .await
+    .unwrap()
+    .quote;
 
-async fn partner_fee_with_not_default_cap_sell_order_test(web3: Web3) {
-    // Fee policy to be overwritten by the partner fee + capped to 0.02
-    let fee_policy = FeePolicyKind::PriceImprovement {
-        factor: 0.5,
-        max_volume_factor: 0.9,
-    };
-    let protocol_fee = ProtocolFee {
-        policy: fee_policy,
-        policy_order_class: FeePolicyOrderClass::Market,
-    };
-    // Without protocol fee:
-    // Expected execution is 10000000000000000000 GNO for
-    // 9871415430342266811 DAI, with executed_surplus_fee = 167058994203399 GNO
-    //
-    // With protocol fee:
-    // Expected executed_surplus_fee is 167058994203399 +
-    // 0.02*(10000000000000000000 - 167058994203399) = 200163717814319331
-    //
-    // Final execution is 10000000000000000000 GNO for 9772701276038844388 DAI, with
-    // executed_surplus_fee = 200163717814319331 GNO
-    //
-    // Settlement contract balance after execution = 200163717814319331 GNO =
-    // 200163717814319331 GNO * 9673987121735421787 / (10000000000000000000 -
-    // 200163717814319331) = 197593222235191520 DAI
-    execute_test(
-        web3.clone(),
-        vec![
-            ProtocolFeesConfig(vec![protocol_fee]).to_string(),
-            "--fee-policy-max-partner-fee=0.02".to_string(),
-        ],
+    let market_price_improvement_uid = services
+        .create_order(&market_price_improvement_order)
+        .await
+        .unwrap();
+    let limit_surplus_order_uid = services.create_order(&limit_surplus_order).await.unwrap();
+    let partner_fee_order_uid = services.create_order(&partner_fee_order).await.unwrap();
+
+    let mut config = vec![
+        "--drivers=test_solver|http://localhost:11088/test_solver".to_string(),
+        "--price-estimation-drivers=test_quoter|http://localhost:11088/test_solver".to_string(),
+    ];
+    config.extend(autopilot_config);
+    services.start_autopilot(None, config);
+
+    tracing::info!("Rebalancing limit order AMM pool.");
```

Comment:
What do you mean by that?

Reply:
Right now you have two blocks of `tracing::info!("Rebalancing ...");` (line ~200 and line ~255). I think it would be cleaner to do all AMM rebalancing right after one another.